dds/DCPS/DomainParticipantFactoryImpl.cpp

```diff
@@ -79,7 +79,7 @@ DomainParticipantFactoryImpl::create_participant(
     return DDS::DomainParticipant::_nil();
   }
 
-  DomainParticipantImpl* dp;
+  DomainParticipantImpl* dp = 0;
```

Comment:
I'm not sure why this is needed, ACE_NEW_RETURN always assigns a value to the pointer.